### PR TITLE
Implement the PHP 7.4  and  methods in addition to the  and  from the…

### PR DESCRIPTION
**Pull request by ideb**
… interface.

### CODE REVIEW

File: lib/SimpleSAML/Session.php [hunk 1]
```diff
@@ -197,17 +197,28 @@ public function setConfiguration(Configuration $config)
...
     }
 
-
     /**
      * Serialize this session object.
      *
      * This method will be invoked by any calls to serialize().
      *
      * @return string The serialized representation of this session object.
+     * @deprecated This method will be removed in SSP 2.0.
```
Comment (tvdijen) [inline lib/SimpleSAML/Session.php:206]: I'm not so sure we can remove this in SSP 2.0 as that will support PHP >= 7.4...    I think we can remove this as soon as the minimum required PHP-version is 8.1?

Comment (ideb) [reply lib/SimpleSAML/Session.php:206]: The __serialize and __unserialize magic methods are available on PHP7.4. Strictly speaking you could remove Serializable right now, but deprecating these non-magic methods gives other projects a chance to remove calls to these methods.

Comment (tvdijen) [reply lib/SimpleSAML/Session.php:206]: You seem to have some experience with this..  Would you mind checking [this PR](https://github.com/simplesamlphp/xml-common/pull/12/files) for me?  It's a generic XML-library which will be responsible for processing SAML XML-structures for SimpleSAMLphp in the near future.. Thank you!

Comment (ideb) [reply lib/SimpleSAML/Session.php:206]: Looks good to me at first glance. Ideally write a unit test for it, so you can check whether a refactoring breaks anything.

**Comment by tvdijen** [comment]
Superseeded by #1597

**Pull request closed without merging**